backward compatibility of restart

aarshkshah1992 opened the pull request:
Closes https://github.com/filecoin-project/lotus/issues/4181.

@hannahhoward Will get the tests in a follow up PR. Let me know what you think.

hannahhoward commented on impl/impl.go:
```diff
@@ -194,7 +194,7 @@ func (m *manager) SendVoucher(ctx context.Context, channelID datatransfer.Channe
 		return errors.New("cannot send voucher for request we did not initiate")
 	}
-	updateRequest, err := message.VoucherRequest(channelID.ID, voucher.Type(), voucher)
+	updateRequest, err := message1_1.VoucherRequest(channelID.ID, voucher.Type(), voucher)
```
one thought about all these renames:

it might make sense, in the root /message directory to alias all of these functions (with a variable) i.e in
`message/message.go`:
```
import (
   "github.com/filecoin-project/go-data-transfer/message/message1_1"
)

var VoucherRequest = message1_1.VoucherRequest
var NewResponse = message1_1.NewResponse
var NewRequest = message1_1.NewRequest
...
```

That way if there is a message1_2 at some point we won't have to make another change like this.


 -- that way, we can avoid another change of this source

aarshkshah1992 replied:
This is done.

hannahhoward reviewed (changes requested):
Some non-blocking suggestions but:
1. Make sure to use --map-encoding for 1_1 cbor-generation
2. Add check for restart message types when down converting 1_1 to 1_0

aarshkshah1992 commented on network/libp2p_impl.go:
```diff
@@ -96,11 +98,16 @@ func (dtnet *libp2pDataTransferNetwork) SendMessage(
 	outgoing datatransfer.Message) error {
 
-	s, err := dtnet.openStream(ctx, p, ProtocolDataTransfer)
+	s, err := dtnet.openStream(ctx, p, datatransfer.ProtocolDataTransfer1_1, datatransfer.ProtocolDataTransfer1_0)
```
This is done.

aarshkshah1992 commented on message/message1_1/transfer_request.go:
```diff
@@ -0,0 +1,166 @@
+	switch targetProtocol {
+	case datatransfer.ProtocolDataTransfer1_1:
+		return trq, nil
+	case datatransfer.ProtocolDataTransfer1_0:
```
This is done.

aarshkshah1992 commented:
@hannahhoward Have addressed your review and have added tests.

The only thing remaining is using `--map-encoding` as upgrading the`cbor-gen-for` library and changing to that encoding fails the cbor generation. Will discuss this with you.

hannahhoward reviewed (approved):
LGTM
